src/main/java/net/dv8tion/jda/api/interactions/commands/LocalizationMap.java

```diff
@@ -0,0 +1,178 @@
+     *
+     * @return The parsed LocalizationMap instance, which can be further configured through setters
+     */
+    @Nonnull
+    public static LocalizationMap fromData(@Nonnull Consumer<String> checkConsumer, @Nonnull DataObject data)
+    {
+        return new LocalizationMap(checkConsumer, data);
+    }
+
+    /**
+     * Parses the provided localization property, in the serialization, back into an LocalizationMap instance.
+     * <br>This is the reverse function for {@link #toData()}.
+     *
+     * @param  json
+     *         The serialized {@link DataObject} containing the localization map
+     * @param  localizationProperty
+     *         The name of the property which represents the localization map
+     * @param  checkConsumer
+     *         The check to run on every localization entry insertion
+     *
+     * @throws net.dv8tion.jda.api.exceptions.ParsingException
+     *         If the serialized object is missing required fields
+     *
+     * @return The parsed LocalizationMap instance, which can be further configured through setters
+     */
+    @Nonnull
+    public static LocalizationMap fromProperty(@Nonnull DataObject json, @Nonnull String localizationProperty, @Nonnull Consumer<String> checkConsumer) {
+        return json.optObject(localizationProperty)
+                .map(data -> fromData(checkConsumer, data))
+                .orElse(new LocalizationMap(checkConsumer));
+    }
+
+    @Nonnull
+    @Override
+    public DataObject toData()
+    {
+        final DataObject data = DataObject.empty();
+        map.forEach((locale, localizedString) -> data.put(locale.toLanguageTag(), localizedString));
+        return data;
+    }
+
+    /**
+     * Sets the given localized string to be used for the specified locales.
+     *
+     * @param  localizedString
+     *         The localized string to use
+     * @param  locales
+     *         The locales on which to apply the localized string
+     */
+    public void setTranslations(@Nonnull String localizedString, @Nonnull Locale... locales)
+    {
+        checkConsumer.accept(localizedString);
+        for (Locale locale : locales)
+            map.put(locale, localizedString);
+    }
+
+    /**
+     * Adds all the translations from the supplied map into this LocalizationMap.
+     *
+     * @param  map
+     *         The map containing the localized strings
+     */
+    public void putTranslations(@Nonnull Map<Locale, String> map)
+    {
+        for (String localizedString : map.values())
+            checkConsumer.accept(localizedString);
+        this.map.putAll(map);
+    }
+
+    /**
+     * Sets the given localized string to be used for the specified locale.
+     *
+     * @param  localizedString
+     *         The localized string to use
+     * @param  locale
+     *         The locale on which to apply the localized string
+     */
+    public void setTranslation(@Nonnull String localizedString, @Nonnull Locale locale) {
```

Comment:
These should all be the same name, differing only by inputs. We don't need a `putX` and a `setX`.